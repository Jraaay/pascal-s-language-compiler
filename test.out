{
    "ast": null,
    "symbolTable": {
        "constants": [],
        "variables": [],
        "subFunc": []
    },
    "error": [
        {
            "code": "B-01",
            "info": {
                "line": 7,
                "value": "a2",
                "lexpos": 115
            }
        }
    ]
}