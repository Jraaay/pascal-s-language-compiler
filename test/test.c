#include<stdio.h>
#include<stdbool.h>

/* Error,  Parser gives no AST. */