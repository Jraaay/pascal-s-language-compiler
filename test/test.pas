Program example(input, output);

Const 
    a = 5; 
    b = -6; 
    c = 't';

Var 
    x, y, d : integer;
    e : array [1..50] of char;
    f : record
        f1 : real;
        f2 : array [1..50, 1..100] of boolean;
    End;

Function gcd(var a, b : integer) : integer;
Begin
  If b = 0 Then gcd := a
  Else gcd := gcd(b, a Mod b)
End;

Begin
read(x,y);
write(x,y);
while x<>y do write(x,y);
gcd(x, y)
End.