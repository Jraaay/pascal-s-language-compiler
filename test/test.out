{
    "ast": null,
    "symbolTable": {
        "constants": [],
        "variables": [],
        "subFunc": []
    },
    "error": [
        {
            "code": "B-01",
            "info": {
                "line": 6,
                "value": "array",
                "lexpos": 86
            }
        }
    ],
    "warning": [],
    "code": "#include<stdio.h>\n#include<stdbool.h>\n\n/* Error,  Parser gives no AST. */"
}